File: fila/Fila Estática/Java/Fila Circular/Enfileiravel.java
public interface Enfileiravel {
    void enfileirar(Object dado); 		//C	enqueue
    Object frente();					//R	front
    void atualizarInicio(Object dado);	//U	update
    void atualizarFim(Object dado);		//U	update  
    Object desenfileirar();				//D dequeue
										//S sort
	boolean estaCheia();
	boolean estaVazia();
	String imprimir();
}

File: fila/Fila Estática/Kotlin/Fila com Pilhas/Enfileiravel.kt
interface Enfileiravel {
    // Metodos principais
    fun enfileirar(dado: Any?)
    fun desenfileirar(): Any?
    fun frente(): Any?
    fun atualizar(dado: Any?)    
    // Metodos auxiliares
    fun estaCheia(): Boolean
    fun estaVazia(): Boolean
    fun imprimir(): String
}